validatedpatterns/retail
# common automatic update

- Split secret counter in two variables
- Validation Schema for common/clustergroup
- - Fixed issues with schema validation in values.schema.json - Modified clustergroup/values.yaml and examples/values-example.yaml to have same sections and values. - Updated tests for  tests/clustergroup-naked.expected.yaml and tests/clustergroup-normal.expected.yaml
- Fixed issues with kubeconform in clustergroup/values.yaml.  Reran make test to generate proper files
- - Updated schema to include all elements in subscriptions - Updated tests associated with clustergroup - Updated values-example.yaml
- applications and subscriptions can be obj and array
- restore values.yaml file to a saner commented default
- Ran values.schema.json through jq to fix indentation issues
- Fix up tests
- Move to draft 0.7 of the schema spec
- Add operatorgroupExcludes which is in examples/industrial-edge-factory.yaml
- namespace is not mandatory in an application
- Fix up some more tests
- add clusterSelector in managedClusterGroups
- Allow clusterRoleYaml to be an array and timeout to be a string
- Add image, tags, extravars to imperative job object
- Add descriptions for SecretStore properties
- move GlobalGit and GitSecret after Main
- Forgot ,
- Move testing targets after the installation ones
- Move some of the oc calls into the legacy target
- Simplify getting the remote git URL associated to the target origin
- Drop legacy-install from common
- Move comments in the right place
- Clean up PHONY targets in Makefile
- Move all testing-related variables closer to the tests target
- Make sure we error out after the legacy removal
- Fix up the main section a bit
- Make subscriptions not mandatory
- Add description to main.git fields
- Drop useCSV from global
- Add description for main.gitops.channel
- Add some descriptions
- Mark useCSV as deprecated
- - Added descriptions for namespaces, subscriptions, managedClusterGroup, applications, and projects
- - Fixed issue with description definition
- drop insecureUnsealVaultInsideCluster entirely
- - Removed insecure boolean from schema file and from values-example.yaml
- Re-ran make tests
- Drop any remains of output_file
- Updated python-version for json schema
- Update python version for ansible-unittest
- Add filter to parse ACM secrets
- Update ESO to 0.7.0
- Only set vaultkeys fact when the vault is sealed
- Default vaultPrefixes to be ["hub"]
- Fix two V2 secret format corner-case bugs
- Add SkipDryRunOnMissingResource=true on ClusterPools, ClusterClaims and ManagedClusterSets
- Add mention of VALUES_SECRET env variable
- Add a get_ini_value() commodity function
- Fix up ini function and add tests
- Add ini_file support
- Relax json schema and drop onMissingValue as required
- Fix up json schema errors
- Fix base64 file upload
- Fix schema json when using HostedArgo
- Simplify hosted argo secret paths
- Add "vault_path" to the parse_acm_secrets filter
- Fix up some pylint warnings
- Set additionalProperties to true for top-level objects
- Fix some validation schema warnings
- Introduce a validate-schema Makefile target
- Do not run validate-schema by default
- helmOverride.value can be both a string and a boolean
- Make sure our readOnly attributes are proper booleans
- Drop managedClusterGroups as a required property
- Drop 'pattern' as a required property
- Run check-json schema for all yaml files under examples/
- Allow subscriptions to be null
- Add prereq check to the operator-deploy
- Vault delete task is unused, remove it
- Drop pki init as it is unused
- Drop push_secrets.yaml playbook as it is unused
- Make sure we error out with a proper message if python-kubernetes is missing
- Make prerequisited output checking more consistent
- Prettify the output for checking prerequisites
- Extend framework to allow for ESO on ACM-imported clusters
- Add a global policy in the vault configuration
- Make sure we use specialized vars for spokes
- Switch to quay.io/hybridcloudpatterns/utility-container
- Upgrade ansible-lint action
- Address a bunch of ansible-lint warnings
- Upgrade ESO to v0.7.2
- Add acm policy to push HUB's CA to managed clusters
- Point golang-external-secrets to the secret injected by ACM
- Drop imperative job to fetch ACM HUB CA as we use ACM to push that out to clusters
- Fix up tests
- Loosen grep regex to catch compound/alias Makefile targets
- Add pattern-home mount to allow resolution of ~ in files
- Better help output
- Stop outputting help for deprecated commands
- Allow additional properties for GlobalGit
- Print which files we are using to parse values secrets
- Drop SSH_SOCK_MOUNT from pattern-util.sh
- Add /values-<platform>.yaml and /values-<platform>-<clusterversion>.yaml support
- Fix up tests from new common/
- Clean up Makefile after common rebase
- Add common-* tests now that we test all charts in the pattern repo
- Fix up Makefile help and more
